app/assets/javascripts/tylium/pages/activities.js

```diff
@@ -29,4 +29,20 @@ document.addEventListener('turbolinks:load', function() {
       }
     });
   }
+
+  $(function() {
+    $('[data-behavior~=activity-filters] select').change(function () {
+      $(this).closest('form').submit();
+    });
+
+    $('[data-behavior~=activity-filters] input[type="date"]').change(function () {
+      const dateValue = $(this).val();
+      const periodStart = $('[data-behavior~=period-start]').attr('min');
+      if (dateValue !== '' && !isNaN(Date.parse(dateValue))) {
+        if (Date.parse(dateValue) >= Date.parse(periodStart)) {
+          $(this).closest('form').submit();
+        }
```

Comment:
Shouldn't we also check if periodEnd >= periodStart?

Reply:
The reason we have this check is to ensure that the form won't be submitted with an incomplete starting date. We wanted to find a way to make sure that a user can input a date using the keyboard as well as the mouse. 

However, through the keyboard, the form was being submitted as long as we were hitting the year's part(dd/mm/**yyyy**) of the input. To prevent this from happening we are checking that a starting date is complete before submitting the form. This edge case scenario only happens with periodStart since the periodEnd is always filled.